.travis.yml

```diff
@@ -1,2 +1,5 @@
 language: rust
 rust: nightly
+script:
+    - cargo test
+    - cd examples calc && cargo build
```

Comment:
The example doesn't have its own tests? Would that be helpful documentation to the user?

Reply:
I think the fact that it's just `cargo build` and not `cargo test` is indication enough.

Reply:
Sorry, I meant would adding a test or two help the user understand the API better? Now I've read the whole PR I'm not so sure...